### ISSUE TITLE
Application::getCachedPackagesPath fires before environment variables load

### ISSUE THREAD

**Issue by JasonTheAdams** [laravel/framework#28347]
- Laravel Version: 5.8.14
- PHP Version: 7.2.0
- Database Driver & Version: MySQL 5.7

### Description:
I'm trying to deploy Laravel on a Google App Engine Standard 7.2 instance. A trick in doing this is that files cannot be written anywhere in the application directory, but can be written in `/tmp`. This is handled by using the appropriate environment variables.

One such variable is `APP_PACKAGES_CACHE` [defined in the Application class](https://github.com/laravel/framework/blob/5.8/src/Illuminate/Foundation/Application.php#L896). This is used in the `php artisan package:discover` command to cache the services and packages.

The problem is that by the time the `Application::getCachedPackagesPath` method gets fired, the `$_ENV` array hasn't been populated from the `.env` file yet.

By contrast, the `APP_SERVICES_CACHE` environment variable works just fine. When I stop (via XDEBUG) in the `Application::getCachedServicesPath` line I can see that `$_ENV` is filled, therefore it works.

### Steps To Reproduce:
1. Add the following to the `.env` file:
    ```
    APP_PACKAGES_CACHE=/tmp/cache.packages.php
    APP_SERVICES_CACHE=/tmp/cache.services.php
    ```
2. Run the `php artisan package:discover` command
3. Check the `/tmp` directory — you will see the `cache.services.php` file there, the packages file is still in the default location.

**Comment by driesvints** [comment]
From how I understand this there's not really a way around this because like you say those env variables aren't loaded in yet. You'll have to use a real env variable in your environment?

**Comment by JasonTheAdams** [comment]
@driesvints Thanks for chiming in so quickly.

I suppose I was really curious as to whether this would be considered a bug or not. If the packages must be cached in order for the `$_ENV` file to be populated, then clearly it's just a limitation of the bootstrap process.

I'm curious where you would recommend placing the `putenv('APP_PACKAGES_CACHE', '/tmp/cache.packages.php')` otherwise, as the `bootstrap.app.php` file is fired _after_ the packages are cached?

**Comment by driesvints** [comment]
You said you're using `Google App Engine Standard`. I don't know this service myself but is there no way to defined environment variables anywhere?

**Comment by JasonTheAdams** [comment]
I found another workaround for this. Thanks, @driesvints!

**Comment by crcms** [comment]
@JasonTheAdams Can I ask you what is the other way? I am also experiencing this problem, I need to move all writable directories to the same place, so I have to change the bootstrap / cache path

**Comment by JasonTheAdams** [comment]
I'm sorry, @crcms, I tried to search the commits around that time to give me a clue, but I can't remember what I did. We have it working now. I wish I'd documented in this issue specifically what problem we were having. The packages are built during the build time in App Engine, and at that point the directories are still writable as it's generating the container to be reused — so as long as Laravel isn't trying to create files outside of `/tmp` post-build, it should work find.
